File: server/app/graphql/object_types/report_availability_rate.rb
class ObjectTypes::ReportAvailabilityRate < Types::BaseObject
  graphql_name "ReportAvailabilityRate"

  field :line_id, ID, null: false
  field :line_name, String, null: false
  field :line, ObjectTypes::Line, null: true
  field :date, String, null: true
  field :rate, Float, null: true
  field :prev_date, String, null: true
  field :prev_rate, Float, null: true
end

File: front/api/queries/report/reportAvailabilityRate.ts
import { lineFragment } from "@/api/fragments";
import gql from "graphql-tag";

export default gql`
  query reportAvailabilityRate($search: ReportSearch) {
    reportAvailabilityRate(search: $search) {
      lineId
      lineName
      date
      rate
      prevRate
    }
  }
  ${lineFragment}
`;
